packages/core/src/pipeline/config/stages/deploy/deployStage.js

```diff
@@ -39,6 +39,16 @@ angular
                   clusterServiceProvider.$get().isDeployingArtifact(cluster),
               ),
           },
+          {
+            type: 'imageProviderBeforeType',
+            message: 'You must have a trigger that provides the image metadata  before any deploy stage.',
+            triggerTypes: ['docker', 'jenkins'],
+            skipValidation: (pipeline, stage) => 
+              (stage.clusters || []).every(
+                (cluster) =>
+                  !CloudProviderRegistry.getValue(cluster.provider, 'serverGroup.checkForImageProviders')
+              ),
```

Comment:
What's the purpose of this? (I'm not familiar with this codebase enough, sorry)
btw, you can use `some` instead of `every` and `!`

Reply:
This is to control which cloud providers use this validation. For now this validation is Titus-specific and doesn't apply to anyone else. That is why I only added `checkForImageProviders` flag to the Titus module.